pilot/pkg/model/envoyfilter.go

```diff
@@ -161,6 +169,21 @@ func convertToEnvoyFilterWrapper(local *config.Config) *EnvoyFilterWrapper {
 			Match:     cp.Match,
 			Operation: cp.Patch.Operation,
 		}
+		// For waypoint filters, we need to propagate the targetRefs
+		// to the individual patch wrapper for service match.
+		if cpw.Match != nil && cpw.Match.Context == networking.EnvoyFilter_WAYPOINT {
+			cpw.Hostnames = make([]host.Name, 0, len(localEnvoyFilter.TargetRefs))
+			for _, targetRef := range localEnvoyFilter.TargetRefs {
+				if matchesGroupKind(targetRef, gvk.Service) {
```

Comment:
Matching on hostname is not right. We need to match on the proper GVK to be correct.